### ISSUE TITLE
JWT Decode Error: TypeError: bas...

### ISSUE THREAD

**Issue by Sevyls** [IvanMathy/Boop#133]
Hi 
I like Boop, but JWT Decode does not work for me.

<img width="1159" alt="Bildschirmfoto 2020-07-23 um 09 34 50" src="https://user-images.githubusercontent.com/547842/88262060-c5f5cd00-ccc7-11ea-9d23-5e9baa3fe77b.png">

Here this Example JWT:

```jwt
eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
```

conforms to this Demo JWT: https://jwt.io/#debugger-io

I can't read the error message completely and can't find any way to extend the displayed text.
I also can't locate any detailed log file to provide more details.

Please fix the JWT Decoder.

Kind regards 
Michael

**Comment by IvanMathy** [comment]
Hey there!

That's indeed a bug... I'll push an update sometime today or tomorrow that fixes. 

In the meantime, here's an updated version of the script you can use:

```javascript

/**
  {
    "api":1,
    "name":"JWT Decode (fixed)",
    "description":"Converts JWTs to JSON",
    "author":"Nils Sonemann",
    "icon":"identification",
    "tags":"decode,jwt,token"
  }
**/


const { decode } = require('@boop/base64')

function main(input) {
    var t = input.text
    var jwtParts = t.split(".")
    if(jwtParts.length != 3) {
        input.postError("Invalid Token")
        return
    }

    var header = decode(jwtParts[0]);
    var payload = decode(jwtParts[1]);
    var signature = jwtParts[2];

    try {
        var fullJson = {
            "header": JSON.parse(header),
            "payload": JSON.parse(payload),
            "signature": signature
        }

        // Prettyprint the JSOM
        input.text = JSON.stringify(fullJson, null, 2);
    } catch(err) {
        input.postError("Error while parsing JSON")
    }
}


```

**Comment by IvanMathy** [comment]
Fixed in release 1.2.1
